src/features/machine/polarMachine.js

```diff
@@ -0,0 +1,210 @@
+
+  if (deltaAngle > Math.PI) {
+    deltaAngle -= 2.0 * Math.PI
+  }
+  if (deltaAngle < 0.0) {
+    resolution *= -1.0
+  }
+
+  let tracePoints = []
+  for (let step = 0; step < (deltaAngle/resolution) ; step++) {
+    tracePoints.push(Victor(size * Math.cos(resolution * step + startAngle),
+                            size * Math.sin(resolution * step + startAngle)))
+  }
+  return tracePoints
+}
+
+export default class PolarMachine extends Machine {
+  constructor(vertices, settings) {
+    super()
+    this.vertices = vertices
+    this.settings = settings
+  }
+
+  addEndpoints() {
+    const maxRadius = this.settings.maxRadius
+
+    if (this.settings.polarStartPoint !== 'none') {
+      if (this.settings.polarStartPoint === 'center') {
+        this.vertices.unshift(new Victor(0.0, 0.0))
+      } else {
+        const first = this.vertices[0]
+        const scale = maxRadius / first.magnitude()
+        const startPoint = Victor.fromObject(first).multiply(new Victor(scale, scale))
+        this.vertices.unshift(new Victor(startPoint.x, startPoint.y))
+      }
+    }
+
+    if (this.settings.polarEndPoint !== 'none') {
+      if (this.settings.polarEndPoint === 'center') {
+        this.vertices.push(new Victor(0.0, 0.0))
+      } else {
+        const last = this.vertices[this.vertices.length-1]
+        const scale = maxRadius / last.magnitude()
+        const endPoint =  Victor.fromObject(last).multiply(new Victor(scale, scale))
+        this.vertices.push(new Victor(endPoint.x, endPoint.y))
+      }
+    }
+
+    return this
+  }
+
+  // Finds the nearest vertex that is in the bounds of the circle. This will change the
+  // shape. i.e. this doesn't care about the line segment, only about the point.
+  nearestVertex(vertex) {
+    const size = this.settings.maxRadius
+
+    if ( vertex.length() > size) {
+      const scale = size / vertex.length()
+      return vertex.multiply(new Victor(scale, scale))
+    } else {
+      return vertex
+    }
+  }
+
+  // Returns the distance along the perimeter between two points.
+  perimeterDistance(v1, v2) {
+    const startAngle = angle(v1)
+    const endAngle = angle(v2)
+    let deltaAngle = Math.abs(endAngle - startAngle)
+
+    if (deltaAngle > Math.PI) {
+      deltaAngle -= 2.0 * Math.PI
+    }
+
+    return Math.abs(deltaAngle) * this.settings.maxRadius
+  }
+
+  // Returns points along the circle from the start to the end, tracing a circle of radius size.
+  tracePerimeter(start, end) {
+    return traceCircle(start, end, this.settings.maxRadius)
+  }
+
+  // Returns whether a given path lies on the perimeter of the circle.
+  onPerimeter(v1, v2, delta=.001) {
+    let rm = Math.pow(this.settings.maxRadius, 2)
+    let r1 = Math.pow(v1.x, 2) + Math.pow(v1.y, 2)
+    let r2 = Math.pow(v2.x, 2) + Math.pow(v2.y, 2)
+    let d = this.perimeterDistance(v1, v2)
+
+    return (r1 >= rm - delta && r2 >= rm - delta) && d < 15
```

Comment:
Where does the 15 come from? It shouldn't ever get triggered, right? The d is in radians, [-pi,pi]?